pkg/apis/monitoring/v1/prometheusrule_types.go

```diff
@@ -86,6 +86,8 @@ type Rule struct {
 	Expr intstr.IntOrString `json:"expr"`
 	// Alerts are considered firing once they have been returned for this long.
 	For Duration `json:"for,omitempty"`
+	// KeepFiringFor is the duration for which alerts will continue firing after the condition that triggered it has cleared.
+	KeepFiringFor Duration `json:"keep_firing_for,omitempty"`
```

Comment:
```suggestion
	KeepFiringFor Duration `json:"keepFiringFor,omitempty"`
```

Reply:
I think `PrometheusRule` was meant to be a simple wrapper around prometheus rules that don't need data manipulation. As such we previously were using the same format as in prometheus/thanos (as seen with `partial_response_strategy`). If we went with `keepFiringFor` then every user would need to mangle data before using PrometheusRule.

Reply:
Honestly I dislike that `partial_response_strategy` isn't `partialResponseStrategy` as it breaks the well-known Kubernetes naming conventions :)
I discussed some ideas to ease the translation from native Prometheus rules to custom resources in https://github.com/prometheus-operator/prometheus-operator/issues/5035.

Reply:
I agree that `partialResponseStrategy` would look nicer and be more in line with other names. However, I think this is not viable and leads to poor UX when we count in how `PrometheusRule` is used and what flow we established over the years. IMHO in the majority of cases, `PrometheusRule` is used as a simple wrapper on data generated elsewhere in prometheus format as this allows easy use of `promtool` for validation before putting data into a cluster (for example in CI/CD system). After data is generated it takes just a simple bash script (like [this one](https://github.com/paulfantom/dotfiles/blob/master/dots/functions#L173-L189)) or a simple jsonnet wrapping (like [here](https://github.com/prometheus-operator/kube-prometheus/blob/main/jsonnet/kube-prometheus/components/alertmanager.libsonnet#L92-L104)) to include this into a cluster (here it is operator job to validate again using since not everyone is doing validation in CI). This allows for a straightforward UX that doesn't require any specialized tools or data manipulation for the small cost of using snake_case in the CR.

Reply:
It shouldn't be too hard to adjust the jsonnet  wrapper but yeah, I see your point. I don't have other practical argument to oppose and `partial_response_strategy` is already a precedence so :shrug: for me.

Reply:
Even in the case of jsonnet wrapper user would need to create a function to iterate over all rules and rule groups to swap the name of one particular field if it exists. It's not the hardest thing to do but also not the simplest one and definitely not as simple as the current solution.

Reply:
it's a one-time change to jsonnet so really not complicated but like I said, I won't object to `keep_firing_for` ;)